elide-core/src/main/java/com/yahoo/elide/core/EnforceJoinFilterExpressionVisitor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import com.yahoo.elide.annotation.ReadPermission;
+import com.yahoo.elide.core.exceptions.ForbiddenAccessException;
+import com.yahoo.elide.core.filter.FilterPredicate;
+import com.yahoo.elide.core.filter.expression.AndFilterExpression;
+import com.yahoo.elide.core.filter.expression.FilterExpressionVisitor;
+import com.yahoo.elide.core.filter.expression.NotFilterExpression;
+import com.yahoo.elide.core.filter.expression.OrFilterExpression;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Enforce read permission on filter join
+ */
+public class EnforceJoinFilterExpressionVisitor implements FilterExpressionVisitor<Boolean> {
```

Comment:
Can we add a Unit test for this class to test the different visitor methods?

Reply:
Can try.